Download/install aws-cfn-bootstrap-py3-latest.tar.gz for all OSes besides Amazon Linux

hanwen-pcluste opened the pull request:
### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.

The latest release (version 0.6.0) of [pystache](https://pypi.org/project/pystache/) removed the support for Python2. This package is a dependency of aws-cfn-bootstrap-latest.tar.gz.

The latest (version 1.4-34) [aws-cfn-bootstrap-latest.tar.gz](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/releasehistory-aws-cfn-bootstrap.html#releasehistory-aws-cfn-bootstrap-v1) became incompatible with Python2 after the pystache update, and it is recommended in [CloudFormation helper scripts reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-helper-scripts-reference.html) to use the latest installation package for Python3.
* Link to impacted open issues.

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
All tests in pcluster3.yaml have been passed
* Describe the added/modified tests.

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

demartinofra commented on tests/integration-tests/tests/ad_integration/test_ad_integration.py:
```diff
@@ -243,7 +243,7 @@ def _create_nlb_stack(cfn_stacks_factory, request, directory_stack, region, test
     )
     logging.info("Creating stack %s", nlb_stack_name)
     # TODO: don't hardcode this ARN
-    certificate_arn = "arn:aws:acm:us-east-1:447714826191:certificate/a17e8574-0cea-4d4c-8e79-a8ebb60f6f47"
```
probably you did not mean to change this?

hanwen-pcluste replied:
Sorry

hanwen-cluster committed:
Download/install aws-cfn-bootstrap-py3-latest.tar.gz for all OSes besides Amazon Linux

The latest release (version 0.6.0) of [pystache](https://pypi.org/project/pystache/) removed the support for Python2. This package is a dependency of aws-cfn-bootstrap-latest.tar.gz.

The latest (version 1.4-34)  [aws-cfn-bootstrap-latest.tar.gz](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/releasehistory-aws-cfn-bootstrap.html#releasehistory-aws-cfn-bootstrap-v1) became incompatible with Python2 after the pystache update, and it is recommended in [CloudFormation helper scripts reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-helper-scripts-reference.html) to use the latest installation package for Python3.

Signed-off-by: Hanwen <hanwenli@amazon.com>
